openBCISample.js

```diff
@@ -61,6 +60,35 @@ module.exports = {
             stopByte: dataBuf[numberOfBytes - 1] // byte
         }
     },
+    convertSampleToPacket: function(sample) {
+        var packetBuffer = new Buffer(k.OBCIPacketSize);
+        packetBuffer.fill(0);
+
+        // start byte
+        packetBuffer[0] = k.OBCIByteStart;
+
+        // sample number
+        packetBuffer[1] = sample.sampleNumber;
+
+        // channel data
+        for (var i = 0; i < k.OBCINumberOfChannelsDefault; i++) {
+            var threeByteBuffer = this.floatTo3ByteBuffer(sample.channelData[i]);
```

Comment:
Yup!!

AJ Keller
Founder 
Push The World
843-471-0039




On Fri, Feb 5, 2016 at 1:44 PM -0800, "Teon L Brooks" <notifications@github.com> wrote:












In openBCISample.js:
> @@ -61,6 +60,35 @@ module.exports = {
>              stopByte: dataBuf[numberOfBytes - 1] // byte
>          }
>      },
> +    convertSampleToPacket: function(sample) {
> +        var packetBuffer = new Buffer(k.OBCIPacketSize);
> +        packetBuffer.fill(0);
> +
> +        // start byte
> +        packetBuffer[0] = k.OBCIByteStart;
> +
> +        // sample number
> +        packetBuffer[1] = sample.sampleNumber;
> +
> +        // channel data
> +        for (var i = 0; i < k.OBCINumberOfChannelsDefault; i++) {
> +            var threeByteBuffer = this.floatTo3ByteBuffer(sample.channelData[i]);



floatTo3Byte? is this to go back the other way for the simulator code?



—
Reply to this email directly or view it on GitHub.